eng/tools/specs-model/.prettierrc.mjs

```diff
@@ -0,0 +1,13 @@
+/**
+ * @see https://prettier.io/docs/en/configuration.html
+ * @type {import("prettier").Config}
+ */
+const config = {
+  tabWidth: 2,
+  semi: false, // No semicolons.
```

Comment:
```suggestion
  semi: true,
```

I think we should align with the format used in the typespec repo as much as possible:

https://github.com/microsoft/typespec/blob/main/.prettierrc.json

In particular, I feel strongly that semicolons should be required.

I also think we should use a single format for all TS code under eng.  After this PR is merged, I can update all the other projects to use the same settings.

Reply:
Totally, I'll copy over typespec prettier. I am just curious why you prefer semicolons so much ;)

Reply:
> Totally, I'll copy over typespec prettier. I am just curious, why you prefer semicolons so much?

Consistency with languages that require them (C, C#, Java).